joomla/joomla-cms
# Added HTTP status code check feature > for the installation #18294

The security file check at the installation could be skipped, if the HTTP status code is not 401 (HTTP authentication required) which means the installation ist http access protected.

I think some web companys like me did work on http access protected development sites. The installation anyway take some steps, so, let's make the life easier and let's skip this if it is not needed.

For this, I have opened this pull request, where the http chek is now implemented.

### Summary of Changes
**Added HTTP status code check,**
for skipping security file if website is http 401 authentication required, which means http access protected.
https://github.com/joomla/joomla-cms/pull/18295/commits/70173fb9409d106c4b9790d87a345c4b2fa4f835

**Outsoureced functionality for the checks to two new methods**
called checkHostSecurity() and checkSecurityFile(). Did also renamed some names called remoteDBFile to securityFile, because i think it is more understandable. Because we are at the InstallationModelDatabase so think its clear which file and it's more associated to the method names. And some improvements in the check order logic. Hope the community like it ;)

The new methodes are called during initialise()

    // Save host checks
    if ($this->checkHostSecurity($options) === false)
    {
        if ($this->checkSecurityFile() === false)
        {
                return false;
        }
    }


### Testing Instructions
- Create a new installation site which is behind an directory protection like http access authentication.
- Go through the installation steps and you shouldn't see the security file check notification.
- And check again, but this time without the http directory protection and the security check should work, the notification should be shown as known ;)

### Documentation Changes Required
https://docs.joomla.org/J3.x:Secured_procedure_for_installing_Joomla_with_a_remote_database

Changes below in ***bold italic*** ↘ 

#### English New Documenation Part:
Starting with Joomla! 3.7.4 the Joomla! Security Strike Team (JSST) implemented additional security checks in the install application in order to protect your web hosting accounts from being overtaken by a remote attacker. In case your database is not on the same server as your website ***and your website is not through a HTTP authentication protected,*** we require an extra check that makes sure you are the owner of the website. 

...

#### German New Documenation Part:
Beginnend mit Joomla! 3.7.4 hat das Joomla! Security Strike Team (JSST) weitere Sicherheitsprüfungen in die Installations-Anwendung eingeführt, um Ihre Webhosting-Konten vor der Übernahme von einem entfernten Angreifer zu schützen. Im Falle, dass die genutzte Datenbank sich nicht auf dem gleichen Server befindet ***und die Website nicht durch eine HTTP-Authentifizierung geschützt ist,*** wird eine separate Prüfung verlangt, um zu versichern, dass Du der Inhaber der Website bist.

...
